mm2src/common/common.rs

```diff
@@ -634,6 +634,28 @@ pub fn var(name: &str) -> Result<String, String> {
 #[cfg(target_arch = "wasm32")]
 pub fn var(_name: &str) -> Result<String, String> { ERR!("Environment variable not supported in WASM") }
 
+#[cfg(not(target_arch = "wasm32"))]
+pub fn block_on_f01<F>(f: F) -> Result<F::Item, F::Error>
+where
+    F: Future,
+{
+    if var("TRACE_BLOCK_ON").map(|v| v == "true") == Ok(true) {
+        let mut trace = String::with_capacity(4096);
+        stack_trace(&mut stack_trace_frame, &mut |l| trace.push_str(l));
+        log::info!("block_on_f01 at\n{}", trace);
+    }
+
+    wio::CORE.0.block_on(f.compat())
+}
+
+#[cfg(target_arch = "wasm32")]
+pub fn block_on_f01<F>(_f: F) -> Result<F::Item, F::Error>
+where
+    F: Future,
+{
+    panic!("block_on_f01 is not supported in WASM!");
+}
+
```

Comment:
No need to handle WASM if you use the suggestion above.

Reply:
[1ebc318](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2220/commits/1ebc3180d581aee9a3ec218f03bb1103f9a10724)